zipkin-collector/src/test/scala/com/twitter/zipkin/collector/BlockingItemQueueTest.scala

```diff
@@ -21,23 +20,48 @@ class BlockingItemQueueTest extends FunSuite {
   }
 
   test("Sleeps on a max queue and waits for the worker to drain") {
-
     val stats:InMemoryStatsReceiver = new InMemoryStatsReceiver()
-    val queue = new BlockingItemQueue[Unit, Unit](10, 1, fallingBehindWorker, 100.millis,
-                                                  100.millis, stats)
+    val queue = new BlockingItemQueue[Unit, Unit](10, 1, fallingBehindWorker, 100.millis, 100.millis, stats)
+    val queueFull = () => stats.counter("queueFull").apply() >= 1
+    val queueSizeZero = () => queue.size() == 0
+    val successSize = () => stats.counter("successes").apply() == 12
+
     // Add 11 and not 10 because the first one that's going to be added will be consumed right away
     assert(Await.result(fill(queue, 11)))
-    assert(Await.ready(queue.add(Item)).poll.get.isReturn)
+
+    // add an item, expect the queueFull stat to get incremented we need this to return because
+    // the blocking queue will block because its full, so we use a future
+    FuturePool.unboundedPool { queue.add(Item) }
+    // poll for when the queueFull counter gets incremented or timeout
+    poll(queueFull)
+    // Items are sitting in the queue blocking and not processing, this unblocks them
+    latch.countDown()
+
+    // Wait for all items to be processed
+    poll(queueSizeZero)
     Await.ready(queue.close())
-    assert(stats.counter("queueFull").apply() >= 1)
-    assert(stats.counter("successes").apply() == 12)
-    assert(queue.size() == 0)
+
+    // All items should be processed
+    poll(successSize)
+    // Final queue should be drained
+    assert(queueSizeZero())
+    assert(queueFull())
```

Comment:
The combination of the above 2 asserts are confusing... the queue has a size zero and it is full?